heron/schedulers/src/java/com/twitter/heron/scheduler/aurora/AuroraScheduler.java

```diff
@@ -187,8 +188,22 @@ public boolean onUpdate(Scheduler.UpdateTopologyRequest request) {
   }
 
   @Override
-  public void addContainers(Set<PackingPlan.ContainerPlan> containersToAdd) {
-    controller.addContainers(containersToAdd.size());
+  public Map<Integer, PackingPlan.ContainerPlan> addContainers(
+      Set<PackingPlan.ContainerPlan> containersToAdd) {
+    // Do the actual containers adding
+    LinkedList<Integer> diffs = new LinkedList<>(
+        controller.addContainers(containersToAdd.size()));
+    Map<Integer, PackingPlan.ContainerPlan> remapping = new HashMap<>();
+    // Do the remapping
```

Comment:
Good to add more comments to illustrate the remapping logic.

Reply:
updated